### ISSUE TITLE
Cannot get reply from PacketSender when using IPv6

### ISSUE THREAD

**Issue by gilfrade** [mfontanini/libtins#283]
Hello, i'm trying make a ping6 example using your library and i cannot catch the reply packet from send_recv(). Using the same code for IPv4 works but not with IPv6. I can see the reply using wireshark.
I'm using a dummy interface.

> void send_packets(PacketSender& sender) {
    // ICMPs are icmp-requests by default
    NetworkInterface::Info info = iface.info();
    IPv6 ip6 = IPv6(addr, info.ipv6_addrs[0].address) / ICMPv6();
    ip6.hop_limit(128);
    ICMPv6& icmp = ip6.rfind_pdu<ICMPv6>();
    icmp.sequence(sequence);
    icmp.identifier(1);
    ttl_map[sequence] = Timestamp().current_time();
    PDU *p = sender.send_recv(ip6,iface);
    recv6_callback(*p);
}

> void recv6_callback(PDU& pdu) {
    // Find IP and ICMP PDUs
    const IPv6& ip = pdu.rfind_pdu<IPv6>();
    const ICMPv6& icmp = pdu.rfind_pdu<ICMPv6>();
    cout << "got reply" << endl;
    if (icmp.type() == ICMPv6::ECHO_REPLY && icmp.sequence() == sequence && ip.src_addr() == addr) {
                results[icmp.identifier()] = ip;
    }
}

**Comment by gilfrade** [comment]
More info. Added the following:
`
         PDU *p = sender.send_recv(ip6,iface);
        if (p == 0) throw pdu_not_found();
`
And it stops where. Is it missing support IPv6 for send_recv()?

**Comment by gilfrade** [comment]
Any word on this?

**Comment by mfontanini** [comment]
Sorry, I tried testing this but I didn't have access to an IPv6 enabled network at the time. I'll give this a try tonight. Which OS are you using?

**Comment by gilfrade** [comment]
Hello, any news?

**Comment by carlosfrade** [comment]
Can you branch this new version so i can try it myself? Thanks

**Comment by gilfrade** [comment]
Can you branch this new version so i can try it myself? Thanks

**Comment by mfontanini** [comment]
Sure, I just pushed the changes I did on branch `develop`. You can try playing around with the socket type being used in PacketSender and see if it works.

**Comment by gilfrade** [comment]
Hello. Have you looked into this issue again?

**Comment by mfontanini** [comment]
No, I haven't looked at this. I have a really busy week so I will definitely not be able to look at this until at least next week.

**Comment by gilfrade** [comment]
Hello. Have you looked at this issue again? Can you tell me your thoughts about what is failing here?
I tried open a raw ipv6 socket from different ways and options but nothing worked. The only thing i saw in other codes was the binding of the socket to a interface, this i did not tried. Regards.

**Comment by mfontanini** [comment]
So I kept trying and I couldn't make it work. However, I did find [this question on SO](https://stackoverflow.com/questions/18055195/getting-the-ipv6-header-as-part-of-the-packet) where someone suggests:

> IPv6 has a different set of options, the equivalent is IPV6_HDRINCL however it does not have widespread support. This implies you may need to drop down to raw packet capture, libpcap and friends.

Looking into the kernel's repo, I find [this commit](https://github.com/torvalds/linux/commit/715f504b118998c41a2079a17e16bf5a8a114885) which adds that socket option. I'm wondering if that would fix it as the code is currently using IP_HDRINCL and it may not be appropriate here. However, that's available as of kernel 4.16 and I currently don't have access to such kernel to test it on. Do you?
